lib/paths/index.ts

```diff
@@ -94,6 +104,17 @@ type UnsafePaths<Type, Options extends Required<PathsOptions>> = Type extends Ty
     : never
   : never;
 
+type AnyArrayIndexAccessor<Type, Options extends Required<PathsOptions>> = Type extends any[]
```

Comment:
You can safely remove type `AnyArrayIndexAccessor` since `anyArrayIndexAccessor` is always a string